docs/content/docs/data-structures/sequence.md

```diff
@@ -0,0 +1,241 @@
+- SharedString for storing and collaborating on a sequence of text.
+- SharedNumberSequence for storing and collaborating on a sequence of numbers.
+- SharedObjectSequence for storing and collaborating on a sequence of JSON-serializable objects.
+
+As the name suggests SharedSegmentSequence, or sequence for short, are made of segments. Segments are the leaf nodes of
+the tree data structure that enables collaboration and backs the sequence. Segments may be split and merged as
+modifications are made to the sequence. Every segment has a length from 1, to the length of the sequence. The length of
+the sequence will be the combined length of all the segments.
+
+When talking about positions in a sequence we use the terms *near*, and *far*. The nearest position in a sequence is 0,
+and the farthest position is its length. When comparing two positions the nearer position is closer to 0, and the
+farther position is closer to the length.
+
+## Using a Sequence
+
+Sequences support three basic operations: insert, remove, and annotate.
+
+Insert operations on the sequence take a single position argument along with the content. This position is inclusive.
+This position can any position in the sequence including 0, and the length of the sequence.
+
+```typescript
+    //   content:
+    // positions:
+
+    // insert text a positions 0
+    sharedString.insertText(0, "hi");
+    //   content: hi
+    // positions: 01
+
+    // insert text at the end position
+    sharedString.insertText(
+        sharedString.getLength(),
+        "!");
+    //   content: hi!
+    // positions: 012
+
+    // insert text at position 2
+    sharedString.insertText(
+        2,
+        " world");
+    //   content: hi world!
+    // positions: 012345678
+```
+
+Remove operations take a start and an end position. The start position is similar to the insert's position, in that is
+can be any position in the sequence and is inclusive. However, unlike insert the start position cannot be the length of
+the sequence, as nothing exists there yet. The end position is exclusive and must be greater than the start, so it can
+be any value from 1 to the length of the sequence.
+
+```typescript
+    //   content: hi world!
+    // positions: 012345678
+
+    // remove the first 3 characters
+    sharedString.removeRange(0, 3);
+    //   content: world!
+    // positions: 012345
+
+    // remove all the characters
+    sharedString.removeRange(0, sharedString.getLength());
+    //   content:
+    // positions:
+```
+
+Annotate operations can add or remove map-like properties to or from content of the sequence. They can store any JSON
+serializable data and have similar behavior to a shared map. Annotate takes a start and end position which work the same
+way as the start and end of the remove operation. In addition to start and end annotate also takes a map-like properties
+object. Each key of the provided properties object will be set on each position of the specified range. Setting a
```

Comment:
You probably want to say here how properties are merged or overwritten (and mentioned what is the default mode)